frontend/src/entities/configuration/dynamic-configuration.tsx

```diff
@@ -115,6 +120,58 @@ export function DynamicConfiguration () {
         // eslint-disable-next-line react-hooks/exhaustive-deps 
     }, [emrApplicationList, applicationConfig]);
 
+    useEffect(() => {
+        // Initialize the notebook instance type selectors with the currently selected options
+        setSelectedNotebookInstanceOptions([]);
+        const selectedNotebookInstances: any[] = [];
+        for (const notebookInstance of applicationConfig.configuration.EnabledInstanceTypes.notebook) {
+            selectedNotebookInstances.push({
+                value: notebookInstance,
+                label: notebookInstance
+            });
+        }
+        setSelectedNotebookInstanceOptions((priorSelectedOptions) => [...priorSelectedOptions, ...selectedNotebookInstances]);
+    }, [applicationConfig.configuration.EnabledInstanceTypes.notebook]);
+
+    useEffect(() => {
+        // Initialize the training job instance type selectors with the currently selected options
+        setSelectedTrainingJobInstanceOptions([]);
+        const selectedTrainingJobInstances: any[] = [];
+        for (const tjInstance of applicationConfig.configuration.EnabledInstanceTypes.trainingJob) {
+            selectedTrainingJobInstances.push({
+                value: tjInstance,
+                label: tjInstance
+            });
+        }
+        setSelectedTrainingJobInstanceOptions((priorSelectedOptions) => [...priorSelectedOptions, ...selectedTrainingJobInstances]);
+    }, [applicationConfig.configuration.EnabledInstanceTypes.trainingJob]);
+
+    useEffect(() => {
+        // Initialize the transform job instance type selectors with the currently selected options
+        setSelectedTransformJobInstanceOptions([]);
+        const selectedTransformInstances: any[] = [];
+        for (const transformInstance of applicationConfig.configuration.EnabledInstanceTypes.transformJob) {
+            selectedTransformInstances.push({
+                value: transformInstance,
+                label: transformInstance
+            });
+        }
+        setSelectedTransformJobInstanceOptions((priorSelectedOptions) => [...priorSelectedOptions, ...selectedTransformInstances]);
+    }, [applicationConfig.configuration.EnabledInstanceTypes.transformJob]);
+
+    useEffect(() => {
+        // Initialize the endpoint instance type selectors with the currently selected options
+        setSelectedEndpointInstanceOptions([]);
+        const selectedEndpointInstances: any[] = [];
+        for (const transformInstance of applicationConfig.configuration.EnabledInstanceTypes.endpoint) {
+            selectedEndpointInstances.push({
+                value: transformInstance,
+                label: transformInstance
+            });
+        }
+        setSelectedEndpointInstanceOptions((priorSelectedOptions) => [...priorSelectedOptions, ...selectedEndpointInstances]);
+    }, [applicationConfig.configuration.EnabledInstanceTypes.endpoint]);
```

Comment:
I think so, we're really only doing this action once at render time

Reply:
So the issue I'm running into is that we update the selectedInstanceOptions value in the `addInstanceType` function, and when we do that I need to update the state variable. 

I tried this out with the Endpoints variable, `selectedEndpointInstanceOptions`, and when I try to update it:
```
selectedEndpointInstanceOptions = detail.selectedOptions;
```

the change isn't reflected in the UI.

Reply:
Even trying to set the list to empty:

```
selectedEndpointInstanceOptions = [];
```

Won't mutate the multiselect component. It seems like it wants a state variable.

Reply:
I think you can stop using `useState`/`useEffect` as discussed and store the instance types directly in the reducer state just like everything else. Then just memoize the generation of the selected options so you're not doing that more than necessary. This should make it all work more inline with how everything else works, remove indirection, and simplify the code.

So as an example it would look like this…

```typescript
const selectedNotebookInstanceOptions = useMemo(() => state.form.configuration.EnabledInstanceTypes.notebook.map((instance) => ({
  value: instance,
  label: instance
}, [state.form.configuration.EnabledInstanceTypes.notebook]);
```

You could even get rid of `addInstanceType` in this scenario and instead just call `setFields` directly.
```typescript
<InstanceTypeMultiSelector
  selectedOptions={selectedNotebookInstanceOptions}
  onChange={({ detail }) => {
    setFields({ 'configuration.EnabledInstanceTypes.notebook': detail.selectedOptions.map((option) => option.value });
  }}
  instanceTypeCategory='InstanceType'
/>
```

Reply:
Nice suggestion, will try it out

Reply:
Implemented it and it works great, changes have been pushed up